cargo-ledger/src/main.rs

```diff
@@ -363,6 +380,46 @@ fn build_app(
         apdu_file_path.to_str().unwrap(),
     );
 
+    if standard_build_tree {
+        // Standard bin name.
+        const STANDARD_BIN_NAME: &str = "app";
+        // Closure to copy file with error message if any.
+        let copy_file_with_err_msg = |src: &Path, dst: &Path| {
+            if let Err(err) = fs::copy(src, dst) {
+                eprintln!("Error copying file to output directory : {}", err);
+            }
+        };
+        // Closure to copy generated binaries to standard path.
+        let copy_binaries_to_standard_path = |dir: &Path| {
+            let elf_dest = dir.join(format!("{}.elf", STANDARD_BIN_NAME));
+            let hex_dest = dir.join(format!("{}.hex", STANDARD_BIN_NAME));
+            let apdu_dest = dir.join(format!("{}.apdu", STANDARD_BIN_NAME));
+            println!(
+                "Copying generated binaries to standard directory : {:?}",
+                dir
+            );
+            if let Err(err) = fs::create_dir(&dir) {
+                eprintln!(
+                    "Error creating standard device build directory : {}",
+                    err
+                );
+            }
+            copy_file_with_err_msg(&exe_path, &elf_dest);
+            copy_file_with_err_msg(&hex_file_abs, &hex_dest);
+            copy_file_with_err_msg(&apdu_file_path, &apdu_dest);
+        };
+        // Copying binaries to standard bin directory.
+        let bin_dir = PathBuf::new().join(current_dir).join("bin");
+        copy_binaries_to_standard_path(&bin_dir);
+        // Copying binaries to standard device build directory.
+        let target_build_dir = PathBuf::new()
+            .join(current_dir)
+            .join("build")
+            .join(device.as_ref_for_std_tree())
+            .join("bin");
+        copy_binaries_to_standard_path(&target_build_dir);
```

Comment:
Copy all binaries into a C-SDK like build tree, so do not understand about the `/bin` folder above...

Reply:
We have both in the C SDK actually, the `bin` directory with the binaries from the latest `make` (files are overwritten every time, whatever the target selected) and the `build/<target>/bin/` tree that keeps the files of every target we built the app for.

Reply:
Some of our tools might use the `bin` and others the `build/<target>/bin`

Reply:
So for Rust, I am afraid that it is quite redundant and confusing. Having three (!) different folders with binaries, it is a lot...I was understanding that the `build` folder is required for other tools (ragger for instance). In Rust, both C SDK folders `bin` and `build` folders are merged into `target` folder as it already contains all the binaries for the different targets and it is updated at every build for the current target. So I do not see any reason to have the `bin` folder.

Reply:
Could we list the tools that need either `bin` or `build` folders ?

Reply:
It will never be an issue as it is opt-in only (enabled by the flag, optionally). What's more it will only ever be used by : 

* Our team internally (we know what we do),
* External developers who know what they do.

If it is used by people who don't know what they do, yes it might get confusing, but I don't really mind.

Also, we should stop thinking about C apps vs Rust apps. We have apps, and they should all integrate with our tooling in the same way. I personally don't want to lose time looking over every piece of code that our apps integrate with and justify our team choices over the past years that led to having `bin`  and `build`. It is the state of things today, and all apps should be able to provide this.

@yogh333 if you want to spend time doing the archaeology, go ahead. If you want to change how things are done, please do so as well, but do it for all tools and apps (so please update the C SDK as well).

Reply:
Hi, 

 I just would like to understand why we have to bend cargo-ledger to match with some obscure C-SDK constraints. I like understanding as, in general, it helps to take good and future-proof decision. I do not think it is a waste of time.
And yes we could change how things are done if it is better, but for this we need to **understand**.
I am still ok with the opt flag, as we agreed collectively.
Last but not least, I think that we still still a `cargo ledger clean` if our `cargo ledger build` creates some aditionnal non native Rust build folders.

Reply:
Agree about `cargo ledger clean`